src/libs/actions/User.ts

```diff
@@ -12,83 +12,93 @@ import * as ReportActionsUtils from '@libs/ReportActionsUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {FrequentlyUsedEmoji} from '@src/types/onyx';
+import type Login from '@src/types/onyx/Login';
+import {OnyxServerUpdate} from '@src/types/onyx/OnyxUpdatesFromServer';
+import type OnyxPersonalDetails from '@src/types/onyx/PersonalDetails';
+import ReportAction from '@src/types/onyx/ReportAction';
 import * as Link from './Link';
 import * as OnyxUpdates from './OnyxUpdates';
 import * as PersonalDetails from './PersonalDetails';
 import * as Report from './Report';
 import * as Session from './Session';
 import redirectToSignIn from './SignInRedirect';
 
-let currentUserAccountID = '';
+type CustomStatus = {text: string; emojiCode: string; clearAfter?: string};
+type BlockedFromConciergeNVP = {expiresAt: number};
+
+let currentUserAccountID = -1;
 let currentEmail = '';
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
-        currentUserAccountID = lodashGet(val, 'accountID', -1);
-        currentEmail = lodashGet(val, 'email', '');
+    callback: (value) => {
+        currentUserAccountID = value?.accountID ?? -1;
+        currentEmail = value?.email ?? '';
     },
 });
 
-let myPersonalDetails = {};
+let myPersonalDetails: OnyxPersonalDetails | Record<string, never> = {};
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-    callback: (val) => {
-        if (!val || !currentUserAccountID) {
+    callback: (value) => {
+        if (!value || currentUserAccountID === -1) {
             return;
         }
 
-        myPersonalDetails = val[currentUserAccountID];
+        myPersonalDetails = value[currentUserAccountID];
     },
 });
 
 /**
  * Attempt to close the user's account
  *
- * @param {String} message optional reason for closing account
+ * @param message optional reason for closing account
  */
-function closeAccount(message) {
+function closeAccount(message: string) {
     // Note: successData does not need to set isLoading to false because if the CloseAccount
     // command succeeds, a Pusher response will clear all Onyx data.
-    API.write(
-        'CloseAccount',
-        {message},
-        {
-            optimisticData: [
-                {
-                    onyxMethod: Onyx.METHOD.MERGE,
-                    key: ONYXKEYS.FORMS.CLOSE_ACCOUNT_FORM,
-                    value: {isLoading: true},
-                },
-            ],
-            failureData: [
-                {
-                    onyxMethod: Onyx.METHOD.MERGE,
-                    key: ONYXKEYS.FORMS.CLOSE_ACCOUNT_FORM,
-                    value: {isLoading: false},
-                },
-            ],
+
+    type CloseAccountParams = {message: string};
```

Comment:
NAB: We usually put params closers API.write/API.read, under optimisticData, failureData and successData